### ISSUE TITLE
NUnit2021 - false positive for types that override Equals

### ISSUE THREAD

**Issue by PiotrKlecha** [nunit/nunit.analyzers#917]
An object of a type that overrides Equals method can potentially be equal to any other object (depending on implementation) and NUnit itself will actually use the overriden method when testing for equality, but nunit.analyzers reports that the EqualTo constraint will fail (NUnit2021).

An example of the issue (not a real-life one, only simplified to demonstrate):
```cs
public class ObjectWrapper<T>(T wrapped)
{
	public T Wrapped { get; } = wrapped;

	public override bool Equals(object obj) => obj switch
	{
		ObjectWrapper<T> wrapper => Equals(wrapper),
		T wrapped => Equals(wrapped),
		_ => false,
	};

	public override int GetHashCode() => EqualityComparer<T>.Default.GetHashCode(Wrapped);

	protected bool Equals(ObjectWrapper<T> other) => Equals(other.Wrapped);

	protected bool Equals(T otherWrapped) => EqualityComparer<T>.Default.Equals(Wrapped, otherWrapped);
}

[Test]
public void ObjectWrapperEquality()
{
	var value = 10;
	var wrapper = new ObjectWrapper<int>(value);

	Assert.That(wrapper, Is.EqualTo(value));
	Assert.That(value, Is.EqualTo(wrapper));
}
```

Both checks pass, yet both are marked with NUnit2021.

**Comment by manfred-brands** [comment]
This was changed in [nunit](https://github.com/nunit/nunit/pull/4957) after this rule was created, but I agree that the two should match.

I will create a PR to mimic the nunit behaviour.

**Comment by manfred-brands** [comment]
I looked into this, NUnit.Analyzer doesn't warn if your `ObjectWrapper` class implements: `IEquatable<ObjectWrapper<T>>, IEquatable<T>`

I'm afraid that if I add code to test for `Equals(object)` then it will never warn as all classes should override this methods if they do any equality. It doesn't mean that they actually can compare to all types.

Can you live with adding `IEquatable<ObjectWrapper<T>>, IEquatable<T>` to your class?

**Comment by PiotrKlecha** [comment]
Considering all the potential downsides, I agree that it is better to keep the current behavior of the analyzer.

I will use one of the alternatives to fix my test, thank you for the suggestions.
